modules/GenericDataParser/src/server/generic_data_parser.R

```diff
@@ -3953,6 +4063,133 @@ getFormatParameters <- function(rawOnlyFormat, customFormatSettings, inputFormat
+    experimentRowData <- selColumnOrderInfo[experimentRowNum,]
+    experimentRowName = experimentRowData$valueKind
+    experimentRowUnits = experimentRowData$Units
+    experimentRowDataType = experimentRowData$valueType
+    
+    #the experiment column must match at least one endpoint in the protocol to pass 
+    experimentRowMatchesEndpoint = FALSE
+
+    # we need to check if the experiment row data matches any one of the protocol endpoints
+    for (protocolRowNum in seq(1,nrow(protocolEndpointDataFrame))) {
+      # if a protocol value is NA, we consider it a match (since any value is acceptable)
+      # if the experiment values also match the protocol values, we consider it a match
+      # all three (rowNamesMatch, rowUnitsMatch, and rowTypesMatch) must be true
+      protocolRowData <- protocolEndpointDataFrame[protocolRowNum,]
+
+      protocolRowName = protocolRowData$columnName
+      if (is.na(protocolRowName) | experimentRowName == protocolRowName) {
+        rowNamesMatch = TRUE 
+      } else {
+        rowNamesMatch = FALSE
+      }
+
+      protocolRowUnits = protocolRowData$units
+      if (is.na(protocolRowUnits) | experimentRowUnits == protocolRowUnits) {
+        rowUnitsMatch = TRUE 
+      } else {
+        rowUnitsMatch = FALSE
+      }
+
+      protocolRowDataType = protocolRowData$dataType
+      if (is.na(protocolRowDataType) | experimentRowDataType == protocolRowDataType) {
+        rowTypesMatch = TRUE 
+      } else {
+        rowTypesMatch = FALSE
+      }
+
+      #if the experiment column matches one of the endpoints for the protocol, record it and move onto the next column
+      if (rowNamesMatch & rowUnitsMatch & rowTypesMatch) {
+        experimentRowMatchesEndpoint = TRUE
+        break 
+      }
+
+    }
+
+    if (experimentRowMatchesEndpoint == FALSE) {
+      addError(paste0("The result type '", experimentRowName, "' with data type '", experimentRowDataType, "' and units '", experimentRowUnits, "' is not one of the allowed result types for this protocol. Please revise your file or contact an ACAS administrator to update the allowed result types for this protocol."))
+    }
+  }
+
+}
+
+getProtocolStrictEndpointMatching <- function(protocol) {
+  # Check whether the protocol has strict endpoint matching set or not 
+  # Returns true or false
+
+  # TODO - This may cause errors in existing protocols that do not have strict endpoint matching already set
+  # To prevent those errors, we will automatically return false for now. 
+
+  for (lsState in protocol$lsStates) {
+    if (lsState[['lsKind']] == "protocol metadata") {
+      for (value in lsState[['lsValues']]) {
+        if (value[['lsKind']] == "strict endpoint matching" & value[['ignored']] == FALSE) {
+          if (value[['codeValue']] == "true") {
+            return(TRUE)
+          } else {
+            return(FALSE)
+          }   
+        }
+      }
+    }
+  }
+
+  # TODO - remove this after database migration
+  return(FALSE)
```

Comment:
I think it'd make more sense to return `racas::applicationSettings$client.protocol.strictEndpointMatchingDefault` if the protocol doesn't have this value set. It is the default after all.